# R-vs.-Python-Who-is-the-k-mer-King
Many bioinformaticians toggle between Bioconductor/R and Biopython/Pandas. A data-driven benchmark can help others choose the right tool.

ğŸ”¬ R vs. Python: Whoâ€™s the k-mer king?

I took a <b>16S rRNA FASTA file</b> constituting 18 sequences and ran two scripts:
â€¢ R (Biostrings): 33.85 ms
â€¢ Python (BioPython + dict): 21.90 ms

Python won by <b>~1.5Ã—â€”surprising</b>, given Biostrings is C-optimized! The small 4â´=256 k-mer space and Pythonâ€™s lightweight timeit gave it the edge.

ğŸ”¹ R ğŸ¦: built-in genomic tools, concise <br>
ğŸ”¹ Python ğŸ: super-fast for tiny k-mer spaces, merges seamlessly with Pandas/NumPy

Which do you reach for in your pipelines? Drop your thoughts or benchmark tales!

# Directory Structure
benchmark_bio.pynb : Python notebook for benchmarking <br>
benchmark_bio.R : R script for benchmarking <br>
barchart.py : Assisted Bar chart fot Counting Runtimes <br>
sequence.fasta : Test FASTA file used
